src/dsync/dsync.c

```diff
@@ -2201,6 +2679,41 @@ static int dsync_option_output_parse(const char *option, int add_at_head)
     return ret;
 }
 
+/* link_dest doesn't support dirs cross filesystems */
+static int dsync_validate_link_dest(const char *link_dest, const char *dest)
```

Comment:
So that we don't have every process call stat, let's change the body of this routine so that only rank 0 does the work, and then MPI_Bcast the result to everyone.

Reply:
Done